### PR TITLE
Revert "Fix snapshot backtest alignment and trade metrics"

### PR DESCRIPTION
**Pull request by mikkokotila**
Reverts Vaquum/Limen#447.

Scope note:
- This PR is a mechanical revert of #447.
- The only non-code adjustment is release-history sanity: package version `2.3.1` and a matching changelog entry so the revert does not move the project version backwards or erase the published `v2.3.0` history.
- No behavioral follow-up fixes are included here.

Follow-up issues intentionally tracked outside this revert:
- #455 Reconcile regression snapshot normalization between inline and post-run backtests
- #456 Harden legacy `backtest_snapshot` behavior and restore direct coverage
- #457 Restore optional-dependency skip guard in `test_tabpfn` after reverting #447

### CODE REVIEW

File: limen/backtest/backtest_snapshot.py [hunk 1]
```diff
@@ -46,62 +34,37 @@ def backtest_snapshot(df: pd.DataFrame,
...
-    entries = pos & (~pos.shift(1, fill_value=False))
-    trade_runs_count = int(entries.sum())
-    if trades_count_mode == 'bars':
-        trades_count = bars_in_market_count
+    if trades_count_mode == 'runs':
+        entries = pos & (~pos.shift(1, fill_value=False))
+        trades_count = int(entries.sum())
     else:
-        trades_count = trade_runs_count
+        trades_count = int(pos.sum())
```
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/backtest/backtest_snapshot.py:61]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #456 rather than changing behavior inside the revert PR.

File: limen/log/_experiment_backtest_results.py [hunk 2]
```diff
@@ -44,31 +18,19 @@ def _experiment_backtest_results(self: Any, disable_progress_bar: bool = False)
...
-            self.permutation_prediction_performance(i)
-        )
-        result_df = backtest_snapshot(
-            perf_df,
-            execution_lag_bars=1,
-            trades_count_mode='runs',
-        )
+        result_df = backtest_snapshot(self.permutation_prediction_performance(i))
 
         all_rows.append(result_df)
```
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/log/_experiment_backtest_results.py:31]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #455 rather than changing behavior inside the revert PR.

File: limen/backtest/backtest_snapshot.py [hunk 3]
```diff
@@ -1,41 +1,29 @@
 import numpy as np
 import pandas as pd
 
-
 def backtest_snapshot(df: pd.DataFrame,
                      *,
                      pred_col: str = 'predictions',
-                     actual_col: str = 'actuals',
                      open_col: str = 'open',
                      close_col: str = 'close',
```
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/backtest/backtest_snapshot.py:8]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #456 rather than changing behavior inside the revert PR.

File: limen/backtest/backtest_snapshot.py [hunk 4]
```diff
@@ -111,91 +74,31 @@ def backtest_snapshot(df: pd.DataFrame,
...
-    if kelly_wins.size and kelly_losses.size:
-        win_rate = float(kelly_wins.size / kelly_returns.size)
-        loss_rate = float(kelly_losses.size / kelly_returns.size)
-        avg_win = float(kelly_wins.mean())
-        avg_loss = abs(float(kelly_losses.mean()))
-        payout_ratio = avg_win / avg_loss if avg_loss > 0 else np.nan
-        mean_kelly_pct = float((win_rate - (loss_rate / payout_ratio)) * 100.0) if payout_ratio > 0 else np.nan
-    else:
-        mean_kelly_pct = np.nan
+        trade_win_rate_pct = trade_expectancy_pct = _trade_return_mean_pct = np.nan
```
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/backtest/backtest_snapshot.py:95]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #456 rather than changing behavior inside the revert PR.

File: limen/backtest/backtest_snapshot.py [hunk 5]
```diff
@@ -111,91 +74,31 @@ def backtest_snapshot(df: pd.DataFrame,
...
-        trade_return_mean_win_pct = float(trade_wins.mean() * 100.0) if trade_wins.size else np.nan
-        trade_return_mean_loss_pct = float(trade_losses.mean() * 100.0) if trade_losses.size else np.nan
+    eq_gross = (1.0 + R_gross).cumprod()
+    eq_net = (1.0 + R_net).cumprod()
+
+    peak = eq_net.cummax()
+    max_drawdown_pct = float((eq_net / peak - 1.0).min() * 100.0)
+
+    total_return_gross_pct = float((eq_gross.iloc[-1] - 1.0) * 100.0)
+    total_return_net_pct = float((eq_net.iloc[-1]   - 1.0) * 100.0)
```
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/backtest/backtest_snapshot.py:84]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #456 rather than changing behavior inside the revert PR.

File: limen/backtest/backtest_snapshot.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikkokotila) [reply limen/backtest/backtest_snapshot.py:61]: Maintainer decision: keeping #454 a strict mechanical revert of #447. The only non-code adjustment here is version/changelog sanity so the revert does not move the package version backwards or erase the published v2.3.0 history. This thread is intentionally deferred to follow-up issue #456 rather than changing behavior inside the revert PR.

**Review by zero-bang — APPROVED** [review]
Clean mechanical revert of #447, verified against the pre-merge state. The only non-revert delta — version bump to 2.3.1 and the matching changelog entry — is correct and keeps release history monotonic. All CI checks pass. Pre-existing issues restored by the revert are properly tracked in #455, #456, and #457.
